### ISSUE TITLE
[naming-convention] Specificity of `filter` seems low (attempting to create exception)

### ISSUE THREAD

**Issue by G-Rath** [typescript-eslint/typescript-eslint#1478]
I could have misunderstood how this part of `naming-convention` works.

I'm attempting to create an exception for `child_process` so that it doesn't have to be in camel case; however I'd like only variables to have that exception, since `import` isn't checked, so the only allowed use I expect to have in codebases is `const child_process = require('child_process')`.

I'd expect that I could use `filter`, and by extension that `filter` would have a higher specificity, so that it would be evaluated first, thu letting me use it to cancel out other selectors, but it seems that is not the case.

**Repro**

```JS
{
  "rules": {
    "@typescript-eslint/naming-convention": ["error", 
        { selector: 'variable', format: ['camelCase', 'UPPER_CASE'] },
        { selector: 'variable', format: ['snake_case'], filter: 'child_process' },
    ]
  }
}
```

```TS
const child_process = require('child_process');
const snake_case = 1;
```

**Expected Result**
`snake_case` is flagged as not being in camel case, but `child_process` isn't.

**Actual Result**
Both `snake_case` & `child_process` are flagged.

**Additional Info**

**Versions**

Also tried against canary.

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.16.0` |
| `@typescript-eslint/parser`        | `2.16.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.14.1` |
| `npm`                              | `6.13.6` |
